commerce_coordinator/apps/orders/tests/test_views.py

```diff
@@ -0,0 +1,114 @@
+"""
+import logging
+from unittest.mock import MagicMock
+
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+from mock import patch
+from rest_framework.test import APIClient
+
+from commerce_coordinator.apps.orders.tests import ECOMMERCE_REQUEST_EXPECTED_RESPONSE, ECOMMERCE_REQUEST_GET_PARAMETERS
+
+logger = logging.getLogger(__name__)
+
+
+class EcommerceClientMock(MagicMock):
+    """A mock EcommerceApiClient that always returns ECOMMERCE_REQUEST_EXPECTED_RESPONSE."""
+    return_value = ECOMMERCE_REQUEST_EXPECTED_RESPONSE
+
+
+@patch('commerce_coordinator.apps.orders.clients.EcommerceApiClient.get_orders', new_callable=EcommerceClientMock)
+class OrdersEcommerceViewsTests(TestCase):
+    """
+    Tests for Ecommerce order views.
+    """
+    # Disable unused-argument due to global @patch
+    # pylint: disable=unused-argument
+
+    # Use Django Rest Framework client for self.client
+    client_class = APIClient
+
+    # Define test user properties
+    test_user_username = 'test'
+    test_user_email = 'test@example.com'
+    test_user_password = 'secret'
+
+    def setUp(self):
+        """Create test user before test starts."""
+
+        super().setUp()
+        User = get_user_model()
+        User.objects.create_user(self.test_user_username, self.test_user_email, self.test_user_password)
+
+    def tearDown(self):
+        """Log out any user from client after test ends."""
+
+        super().tearDown()
+        self.client.logout()
+
+    def test_ecommerce_view_rejects_post(self, mock_ecommerce_client):
+        """Check POST from authorized user receives a 405 Method Not Allowed."""
+
+        # Login
+        self.client.login(username=self.test_user_username, password=self.test_user_password)
+
+        # Perform POST
+        response = self.client.post(reverse('orders:order_history'), ECOMMERCE_REQUEST_GET_PARAMETERS)
+
+        # Check 405 Method Not Allowed
+        self.assertEqual(response.status_code, 405)
+
+    def test_ecommerce_view_rejects_unauthorized(self, mock_ecommerce_client):
+        """Check unauthorized users querying orders are redirected to login page."""
+
+        # Perform GET without logging in.
+        response = self.client.get(reverse('orders:order_history'), ECOMMERCE_REQUEST_GET_PARAMETERS)
+
+        # Check 302 Found with redirect to login page.
+        self.assertEqual(response.status_code, 302)
+        self.assertIn("/login", response.url)
+
+    def test_ecommerce_view_returns_ok(self, mock_ecommerce_client):
+        """Check authorized user querying orders receives an HTTP 200 OK."""
+
+        # Login
+        self.client.login(username=self.test_user_username, password=self.test_user_password)
+
+        # Perform GET
+        response = self.client.get(reverse('orders:order_history'), ECOMMERCE_REQUEST_GET_PARAMETERS)
+
+        # Check 200 OK
+        self.assertEqual(response.status_code, 200)
+
```

Comment:
Just to confirm: here we're logged in as 'test' but we'll get the response for user 'TestUser' because it's mocked. 

The EcommerceUserOrdersView get function confirms that there is an authenticated user making the request, and then it returns the orders for that authenticated user, but it doesn't actually use the 'username' passed along in the parameters anymore (and so it doesn't need/do any check that the authenticated user matches the user in the parameters) so this works without any trouble. Am I understanding correctly?

Reply:
thanks, Phil!